### PR TITLE
[Serde reflection] improve code and documentation

### PR DESCRIPTION
**Pull request by matbd**
## Motivation

* Improving doc for `serde-reflection`
* Fix a bug in `merge` (now called `unify`) and add unit tests
* Rename crate to match future public name.

## Test Plan

```
cargo x test -p serde-reflection
cargo doc -p serde-reflection --open
```

## Related PRs

https://github.com/libra/libra/pull/3060

### CODE REVIEW

File: common/serde-reflection/Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [package]
-name = "libra-serde-reflection"
+name = "serde_reflection"
```
Comment (bmwill) [inline common/serde-reflection/Cargo.toml:2]: Change this to be `serde-reflection` replacing the `_` with a `-`

File: testsuite/generate-format/Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ structopt = "0.3.12"
 
 libra_types = { path = "../../types", version = "0.1.0", package = "libra-types", features=["fuzzing"] }
 lcs = { path = "../../common/lcs", version = "0.1.0", package = "libra-canonical-serialization" }
-serde_reflection = { path = "../../common/serde-reflection", version = "0.1.0", package = "libra-serde-reflection" }
+serde_reflection = { path = "../../common/serde-reflection", version = "0.1.0" }
```
Comment (bmwill) [inline testsuite/generate-format/Cargo.toml:19]:
```suggestion
serde-reflection = { path = "../../common/serde-reflection", version = "0.1.0" }
```

File: testsuite/generate-format/Cargo.toml [hunk 3]
```diff
@@ -16,7 +16,7 @@ structopt = "0.3.12"
 
 libra_types = { path = "../../types", version = "0.1.0", package = "libra-types", features=["fuzzing"] }
```
Comment (bmwill) [inline testsuite/generate-format/Cargo.toml:17]:
```suggestion
libra-types = { path = "../../types", version = "0.1.0", features=["fuzzing"] }
```

File: common/serde-reflection/Cargo.toml (hunk 1 above)
Comment (matbd) [reply common/serde-reflection/Cargo.toml]: fixed

File: testsuite/generate-format/Cargo.toml (hunk 3 above)
Comment (matbd) [reply testsuite/generate-format/Cargo.toml]: applied, thanks

**Pull request closed without merging**